wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceRuntimes.java

```diff
@@ -249,48 +264,45 @@ public Runtime get(String workspaceId) throws NotFoundException, ServerException
     public void stop(String workspaceId, Map<String, String> options) throws NotFoundException,
                                                                              InfrastructureException,
                                                                              ConflictException {
-
+        RuntimeState state = runtimes.get(workspaceId);
+        if (state == null) {
+            throw new NotFoundException("Workspace with id '" + workspaceId + "' is not running.");
+        }
+        if (!state.status.equals(RUNNING)) {
+            throw new ConflictException(
+                    format("Could not start workspace '%s' because its state is '%s'", workspaceId, state.status));
+        }
+        runtimes.replace(workspaceId, new RuntimeState(state.runtime, STOPPING));
```

Comment:
There is no guarantee that objects will be equal. Some of the fields might change in the process of start, so it really depends on equal method implementation